src/language.cpp

```diff
@@ -0,0 +1,33 @@
+#include "language.h"
+#include <QFile>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QDebug>
+
+Language::Language()
+{
+}
+
+void Language::setTranslationsFromJsonFile(const QString &filePath)
+{
+    QFile file(filePath);
+    if (!file.open(QIODevice::ReadOnly)) {
+        return;
+    }
+    QJsonDocument doc = QJsonDocument::fromJson(file.readAll());
+    file.close();
+    if(doc.isNull()) {
+        qInfo() << "Invalid JSON...\n";
+        return;
+    }
+    auto jsonObj = doc.object();
+    for(auto &key : jsonObj.keys()) {
+        m_translations.insert(key, jsonObj.value(key).toString());
+    }
+    qInfo() << m_translations;
+}
+
+QString Language::getText(const QString &key)
+{
+    return m_translations.value(key);
```

Comment:
We should handle if the translation is not up to date (missing key).
As we will develop kiwix-desktop we will add new text. And translation will takes time to be updated. We must not fail here.

We can fallback to the en verision, or return the key as a text or ...